controllers/hazelcast/hazelcast_status_controller.go

```diff
@@ -59,34 +102,50 @@ func (c *HazelcastClient) start(ctx context.Context, config hazelcast.Config) {
 			c.Unlock()
 		}
 	}(ctx)
+	c.statusTicker = &StatusTicker{
+		ticker: time.NewTicker(time.Minute),
+		done:   make(chan bool),
+	}
+
+	go func(ctx context.Context, s *StatusTicker) {
+		for {
+			select {
+			case <-s.done:
+				return
+			case <-s.ticker.C:
+				c.updateMemberStates(ctx)
+			}
+		}
+	}(ctx, c.statusTicker)
 }
 
 func (c *HazelcastClient) shutdown(ctx context.Context) {
 	c.Lock()
 	defer c.Unlock()
-
 	if c.cancel != nil {
 		c.cancel()
 	}
-
 	if c.client == nil {
 		return
 	}
 	if err := c.client.Shutdown(ctx); err != nil {
 		c.Log.Error(err, "Problem occurred while shutting down the client connection")
 	}
+	if c.statusTicker != nil {
+		c.statusTicker.stop()
+	}
 }
 
 func getStatusUpdateListener(c *HazelcastClient) func(cluster.MembershipStateChanged) {
 	return func(changed cluster.MembershipStateChanged) {
 		if changed.State == cluster.MembershipStateAdded {
 			c.Lock()
-			c.MemberMap[changed.Member.UUID.String()] = changed.Member
+			c.MemberMap[changed.Member.UUID] = newMemberData(changed.Member)
 			c.Unlock()
 			c.Log.Info("Member is added", "member", changed.Member.String())
 		} else if changed.State == cluster.MembershipStateRemoved {
 			c.Lock()
-			delete(c.MemberMap, changed.Member.UUID.String())
+			delete(c.MemberMap, changed.Member.UUID)
 			c.Unlock()
 			c.Log.Info("Member is deleted", "member", changed.Member.String())
 		}
```

Comment:
When I tested the status, I saw that the member statuses don't get enriched until the first ticker comes at 1 min after. Maybe we could add the updateMemberStates function here like the following:

```suggestion
		}
		  c.updateMemberStates(context.Background())
```

Reply:
I realized the same thing, is there any reason behind that? @SeriyBg

Reply:
Yes, its because we only enrich the memberData in updateMemberStates function and it is only called in the ticker goroutine.

Reply:
Good catch, thanks @mtyazici! However, I've implemented it a bit differently, to enrich each member right after the new member event. With this, we will evitate the redundant call to already enriched members